### DISCUSSION TITLE
TextLayer CSS Style/Decoration?

### DISCUSSION THREAD

**Discussion by jfuehner** [visgl/deck.gl#6379] [Q&A] [answered]
Currently the TextLayer has support for the CSS [fontWeight](https://deck.gl/docs/api-reference/layers/text-layer#fontweight) property. 

I have use-cases that need to utilize the [`font`](https://developers.arcgis.com/javascript/latest/api-reference/esri-symbols-Font.html) json object property provided back from ArcGIS Feature Services. The [`font`](https://developers.arcgis.com/javascript/latest/api-reference/esri-symbols-Font.html) object property has `decoration` and `style` properties which the deck TextLayer does not currently have support for.

- [decoration](https://developers.arcgis.com/javascript/latest/api-reference/esri-symbols-Font.html#decoration) **Possible Values**: "underline"|"line-through"|"none"
- [style](https://developers.arcgis.com/javascript/latest/api-reference/esri-symbols-Font.html#style) **Possible Values**: "normal"|"italic"|"oblique"

Thoughts on adding support to the deck TextLayer to support CSS decoration and style properties?

@Pessimistress ^^

**Comment by Pessimistress** [comment, marked as answer]
The font is set here: 

https://github.com/visgl/deck.gl/blob/a42b3dd83462792c08b37b314f1fd55a774b951e/modules/layers/src/text-layer/font-atlas-manager.js#L94

You could theoretically just set `fontWeight` to something like `italic small-caps bold`.

I'm happy to support you if you would open a PR to add the additional props.

**Reply by jfuehner** [reply]
@Pessimistress thanks I'll work on putting together a PR
